meter/rct.go

```diff
@@ -174,10 +171,15 @@ func (m *RCT) batterySoc() (float64, error) {
 
 // queryFloat adds retry logic of recoverable errors to QueryFloat32
 func (m *RCT) queryFloat(id rct.Identifier) (float64, error) {
+	conn, err := rct.NewConnection(m.uri, m.cache)
```

Comment:
as-is this makes the connection cache racy

Reply:
Good point. Will have to look into that.